pandas/tests/indexing/multiindex/test_loc.py

```diff
@@ -695,3 +695,22 @@ def test_loc_getitem_index_differently_ordered_slice_none():
         columns=["a", "b"],
     )
     tm.assert_frame_equal(result, expected)
+
+
+def test_loc_empty_multiindex():
+    # GH#36936
+    arrays = [["a", "a", "b", "a"], ["a", "a", "b", "b"]]
+    index = MultiIndex.from_arrays(arrays, names=("idx1", "idx2"))
+    df = DataFrame([1, 2, 3, 4], index=index, columns=["value"])
+
+    # loc on empty multiindex == loc with False mask
+    empty_multiindex = df.loc[df.loc[:, "value"] == 0, :].index
+    result = df.loc[empty_multiindex, :]
+    expected = df.loc[[False] * len(df.index), :]
```

Comment:
can you construct the expected frame explicityly

Reply:
Constructing an empty DataFrame named `expected` with Multiindex assigns the `expected.index.inferred_type` attribute to `object` by default. 
The value of `result.index.inferred_type` attribute is `string` as inferred from the `index` variable created earlier.
This fails the `tm.assert_frame_equal` test

I used the following code to create an empty dataframe with multiindex:
```python
def test_loc_empty_multiindex():
    # GH#36936
    arrays = [["a", "a", "b", "a"], ["a", "a", "b", "b"]]
    index = MultiIndex.from_arrays(arrays, names=("idx1", "idx2"))
    df = DataFrame([1, 2, 3, 4], index=index, columns=["value"])

    # loc on empty multiindex == loc with False mask
    empty_multiindex = df.loc[df.loc[:, "value"] == 0, :].index
    result = df.loc[empty_multiindex, :]
    index2 = MultiIndex(levels=[[], []], codes=[[], []], names=["idx1", "idx2"])
    expected = DataFrame([], index=index2, columns=["value"], dtype="string")
    expected = expected.astype({"value": "int"})
    tm.assert_frame_equal(result, expected) # this test fails

    # replacing value with loc on empty multiindex
    df.loc[df.loc[df.loc[:, "value"] == 0].index, "value"] = 5
    result = df
    expected = DataFrame([1, 2, 3, 4], index=index, columns=["value"])
    tm.assert_frame_equal(result, expected)
```

If the `index` variable is used directly to create the empty dataframe, it introduces NaN values